apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt

```diff
@@ -264,8 +264,17 @@ open class ChatController(var ctrl: ChatCtrl?, val ntfManager: NtfManager, val a
   private fun startReceiver() {
     Log.d(TAG, "ChatController startReceiver")
     if (receiverStarted) return
-    thread(name="receiver") {
-      GlobalScope.launch { withContext(Dispatchers.IO) { recvMspLoop() } }
+    receiverStarted = true
+    CoroutineScope(Dispatchers.IO).launch {
+      while (true) {
+        val ctrl = ctrl
+        if (ctrl == null) {
+          receiverStarted = false
+          return@launch
```

Comment:
```suggestion
          break
```

Reply:
I understand there are cases for `goto` but let's please avoid it when there are more conventional alternatives...

Reply:
Ok